lib/relay/client/dynamic.go

```diff
@@ -150,15 +151,26 @@ type dynamicAnnouncement struct {
 func relayAddressesOrder(input []string) []string {
 	buckets := make(map[int][]string)
 
+	wg := sync.NewWaitGroup()
+	wg.Add(len(input))
+	results := make(chan urlWithLatency, len(input))
 	for _, relay := range input {
-		latency, err := osutil.GetLatencyForURL(relay)
-		if err != nil {
-			latency = time.Hour
-		}
+		go func(irelay string) {
+			latency, err := osutil.GetLatencyForURL(irelay)
```

Comment:
I remember saying this somewhere at some point but maybe in another PR: I don't think the latency measurement will be correct or relevant when we do hundreds of connects in parallel. Maybe if we are limited to a random subset or relays, or do them in batches, or just take it easy and do it sequentially like now.